### PR TITLE
fix: ensure pydantic models serialize to json-compatible `dict`s.

### LINKED ISSUE #345: API endpoint `/metadata/changesheets:submit` returns HTTP 500 Internal Server Error
### Summary

@sujaypatil96 submitted a changesheet and got an **HTTP 500 Internal Server Error** in response (at ~2:15pm PT on Nov. 1). The full changesheet is [here on Slack](https://nmdc-group.slack.com/archives/C057DTRRC4W/p1698879150835739?thread_ts=1698873412.021529&cid=C057DTRRC4W).

### Investigation so far

@eecavanna checked the container logs and saw the associated Python stack trace ended with:

```py
bson.errors.InvalidDocument: cannot encode object: Url('drs://api.microbiomedata.org/gfs0bs34'), of type: <class 'pydantic_core._pydantic_core.Url'>
```

The full stack trace is [here on Slack](https://nmdc-group.slack.com/archives/C057DTRRC4W/p1698878763222719?thread_ts=1698873412.021529&cid=C057DTRRC4W)

### Background info

`nmdc-runtime` was recently updated to use Pydantic v2 instead of v1 (i.e. `2.4.2` instead of `1.10.13`).

- [PR for dependency bump](https://github.com/microbiomedata/nmdc-runtime/pull/338/files#diff-1c625478493f659da19223e5c886f698276736175298464c0266fc5fde414fbaR553)
- [PR for migration of code](https://github.com/microbiomedata/nmdc-runtime/pull/344/files)

### Other info

There is an [official Pydantic v1-to-v2 migration guide](https://docs.pydantic.dev/2.0/migration/), which includes [a section about URLs](https://docs.pydantic.dev/2.0/migration/#url-and-dsn-types-in-pydanticnetworks-no-longer-inherit-from-str).

CC: @dwinston

### PR DESCRIPTION
**Pull request by dwinston**
add regression test

closes #345

replaces many instances of pydantic-model-to-dict calls (`.dict(...)`) with `.model_dump(mode="json",...)` calls as per pydantic V2 guidance.

### CODE REVIEW

**Review by eecavanna — APPROVED** [review]
I see the related section of the Pydantic v1 to v2 migration guide.

image

- https://docs.pydantic.dev/2.0/migration/#changes-to-pydanticbasemodel

This branch also introduces some changesheet-related tests.

**Comment by eecavanna** [comment]
I recommend holding off on merging this until there is no risk of such a merge complicating the resolution of issue https://github.com/microbiomedata/nmdc-runtime/issues/348.

File: nmdc_runtime/api/endpoints/util.py [hunk 1]
```diff
@@ -448,9 +448,11 @@ def _create_object(
     mdb: MongoDatabase, object_in: DrsObjectIn, mgr_site, drs_id, self_uri
 ):
     drs_obj = DrsObject(
-        **object_in.dict(exclude_unset=True), id=drs_id, self_uri=self_uri
+        **object_in.model_dump(exclude_unset=True, mode="json"),
+        id=drs_id,
+        self_uri=self_uri,
     )
-    doc = drs_obj.dict(exclude_unset=True)
+    doc = drs_obj.model_dump(exclude_unset=True, mode="json")
```
Comment (PeopleMakeCulture) [inline nmdc_runtime/api/endpoints/util.py:455]: I don't think order matters but this is inconsistent with the rest of the codebase

File: nmdc_runtime/site/drsobjects/ingest.py [hunk 2]
```diff
@@ -44,7 +44,11 @@ def claim_metadata_ingest_jobs(
     )
     jobs = []
     while True:
-        rv = client.list_jobs(lr.dict()).json()
+        rv = client.list_jobs(
+            lr.model_dump(
+                mode="json",
+            )
+        ).json()
```
Comment (PeopleMakeCulture) [inline nmdc_runtime/site/drsobjects/ingest.py:51]: do we need to call `.json()` here, if `.model_dump()` already returns a json?

**Review by PeopleMakeCulture — APPROVED** [review]
approved with one none-blocking style comment

**Pull request merged**